polygeist/tools/cgeist/Test/Verification/sycl/parallel_for.cpp

```diff
@@ -0,0 +1,47 @@
+// RUN: clang++ -fsycl -fsycl-targets=spir64-unknown-unknown-syclmlir %s -o %t.out 2>&1 | FileCheck %s --implicit-check-not="{{error|Error}}:"
+
+// RUN: clang++ -fsycl -fsycl-device-only -emit-llvm -fsycl-targets=spir64-unknown-unknown-syclmlir %s -o %t.bc 2>/dev/null
+
+// Test that the LLVMIR generated is verifiable.
+// RUN: opt -verify -disable-output < %t.bc
+
+// Verify that LLVMIR generated is translatable to SPIRV.
+// RUN: llvm-spirv %t.bc
+
+// RUN: llvm-dis %t.bc
+// RUN: cat %t.ll | FileCheck %s --check-prefix=LLVM
+
+// Test that the kernel named `kernel_parallel_for` is generated with the correct signature.
+// LLVM: define weak_odr spir_kernel void {{.*}}kernel_parallel_for(
+// LLVM-SAME:  i32 addrspace(1)* {{.*}}, [[RANGE_TY:%"class.sycl::_V1::range.1"]]* noundef byval([[RANGE_TY]]) {{.*}}, [[RANGE_TY]]* noundef byval([[RANGE_TY]]) {{.*}}, [[ID_TY:%"class.sycl::_V1::id.1"]]* noundef byval([[ID_TY]]) {{.*}})
+
+// Test that all referenced sycl header functions are generated.
+// LLVM-NOT: declare {{.*}} spir_func
+
+#include <sycl/sycl.hpp>
+using namespace sycl;
+
+void host_parallel_for(std::array<int, 1> &A) {
+  auto q = queue{};
+  device d = q.get_device();
+  std::cout << "Using " << d.get_info<info::device::name>() << "\n";
+  auto range = sycl::range<1>{1};
```

Comment:
Maybe we should use a range here that would lead to actual parallel execution, even if with only a few elements, e.g., `range<1>{8}`.